### PR TITLE
Improved Smooth tool

### PR DESCRIPTION
**Pull request by T-F-H**
The old Smooth tool mostly just flattened horizontally.
This new one uses two overlapping gradients to match the angle of the surface it's being applied to.
I would suggest adding a new button for the old smooth tool and calling it "Level" as it is still useful (i.e. for making paths on steep mountains).

github

As I am new to Github and open source, any advice is welcome.

### CODE REVIEW

File: Main.tscn [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[gd_scene load_steps=3 format=2]
```
Comment (Zylann) [inline Main.tscn]: Don't include test files with your commit, only keep changes to the plugin itelf.

**Comment by Zylann** [comment]
While working on accelerating brush operations with GDNative, I considered the addition of an actual blur brush, which properly smooths the landscape. I made it replace the SMOOTH mode, and the old mode is now LEVEL.

**Comment by Zylann** [comment]
The new smooth is now in master, you may give it a try ;)

**Comment by RonanZe** [comment]
Thanx Zylann! I'll try those improvements asap

**Comment by Zylann** [comment]
Closing as there is a box-filter smooth tool now

**Pull request closed without merging**